### PR TITLE
tests: add base tests for commands

### PR DESCRIPTION
**Pull request by michalsn**
**Description**
This PR adds basic tests for commands.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
If anyone has an idea what is going on with rector, I would appreciate any hint.

Locally everything is passing on my machine 🤷‍♂️

**Review by lonnieezell — APPROVED** [review]
Seems like the CLITestCase might be a good test trait to add to the core....

**Comment by michalsn** [comment]
This is rather a temporary solution. Thanks to @kenjis we already have very good Mocking for CLI: https://github.com/codeigniter4/CodeIgniter4/blob/4.5/system/Test/Mock/MockInputOutput.php, but it will be released with v4.5.

**Comment by lonnieezell** [comment]
Ah, ok, nice! I missed that one.

**Comment by kenjis** [comment]
@michalsn 

> Installing rector/rector (0.15.25): Extracting archive

The version is too old.

File: tests/Commands/QueueFailedTest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * @internal
+ */
+final class QueueFailedTest extends CLITestCase
+{
+    /**
+     * @throws Exception
+     */
+    public function testRun(): void
+    {
+        Time::setTestNow('2023-12-19 14:15:16');
```
Comment (kenjis) [inline tests/Commands/QueueFailedTest.php:23]: > Don’t forget to reset the current time after the test with calling it without parameters.
https://codeigniter4.github.io/CodeIgniter4/testing/overview.html#testing-and-time

Comment (michalsn) [reply tests/Commands/QueueFailedTest.php:23]: Thank you.

**Pull request merged**